### ISSUE TITLE
consider renaming construct<T> from "Result returning constructors" tutorial to make<T>

### ISSUE THREAD

**Issue by do-m-en** [ned14/outcome#79]
C++ standard library uses make_* for factory functions (std::make_shared, std::make_unique). I believe that the example would not be any less confusing if construct<T> would follow the same pattern with make<T> (but I could be wrong and make is far less intuitive to understand than construct and it's just my taste of consistency getting in the way...). It's also less to type ;) (some people care about those few characters far too much but hey it's an argument :))

Also on the same note naming the non throwing member function "make" instead of "file" would also remove the need for having a specialization of construct/make free standing function and expect the templated version to always forward the parameters to the T::make function instead of current static_assert error.
And to be honest I believe that make<T> function specialization should almost never do any additional work - default init, other args, ... - than just the forwarding so either keep the free standing function specialization example for such cases or remove it all together as always naming the class construction member function make is good enough for writing templated code:
make<file_handle>() is worse than file_handle::make() and using it in templated class boils down to make<T>() or T::make() so not a real benefit here either.

Also a biased additional suggestion - I never use the two stage construction as shown in the example - construct an object with default constructor and then populate it - and instead always create a private constructor that accepts already initialized members and initialize them in make() function before calling the constructor (if they succeed) but I can see a possible benefit of such two stage construction if you really have allot of such possibly-throwing-during-construction members so considering a change of the example to my preferred way (stage 1: initialize all potentially throwing segments, stage 2: call the constructor with those) is just a biased suggestion that is not as relevant as the things described above (but it shows an alternative where default constructing something first could cause a possibly significant overhead/one or more of the members don't have a default constructor) - but it makes the code more readable since the code guarantees that calling a constructor implies object invariant.

**Comment by ned14** [comment]
As that section of the tutorial points out, there are lots and lots of ways of implementing that, and the tutorial section simply reports what AFIO does. AFIO, incidentally, chose `construct` over `make` to prevent confusion with making a new file, and that rationale doesn't apply to any other library. I chose to report on what AFIO does so people could go study AFIO as a real world application if they needed to, as a lot of people find that section and technique very confusing.

Improving the list of alternative approaches would be great however. I'm not sure of the wisdom of always forwarding parameters to the second stage constructing member function, it produces quite a lot of compiler error message spew, but other than that it's safe. Moving all of the second stage construction into the `make` helper is absolutely another valid approach. Inverting the construction order whereby throwing things first, then the actual object is a novel approach - relies on sane move constructors which is why I disfavoured it, but I can see the advantages.

Let me ponder some on this issue before doing something about it. And thanks for the feedback, you raised some interesting ideas.

**Comment by akrzemi1** [comment]
I have added an entry to FAQ. See commit https://github.com/ned14/outcome/commit/c906272c635e7eea6d555fb375e978fb8b7c92e3. I have also updated the tutorial on `outcome<>` to reflect what we discussed here. I did not mention the performance considerations, as I do not fully understand your explanations. I gather that `result` is fast and `outcome` slower, but it does not fit as rationale for having two types.

@do-m-en: does the rationale in the FAQ make sense to you?
